### PR TITLE
types(router): added router array type for Array RouteConfig

### PR DESCRIPTION
**Pull request by manuelojeda**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Underlying tools
- [x] Other, please describe:
Added a conditional in the generator of router when **TypeScript** is selected, added the routes array as **<Array<RouteConfig>>**, when using the **@vue/composition-api** using the routes without the type may return an error of incompatibility

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Other information:**

### CODE REVIEW

File: packages/@vue/cli-plugin-router/generator/template/src/router/index.js [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
+<%_ if (hasTypeScript) { _%>
+import VueRouter, { RouteConfig } from 'vue-router'
+<%_ } else { _%>
 import VueRouter from 'vue-router'
+<%_ } _%>
 import Home from '../views/Home.vue'
 
-Vue.use(VueRouter)
+<%_ if (hasTypeScript) { _%>
+const routes = <Array<RouteConfig>>[
```
Comment (sodatea) [inline packages/@vue/cli-plugin-router/generator/template/src/router/index.js]: As far as I can see, only this line needs to be modified for typescript projects. So no need to  wrap the whole block in the `if` condition.

Comment (sodatea) [reply packages/@vue/cli-plugin-router/generator/template/src/router/index.js]: Better use type annotation than type casting:
```suggestion
const routes: Array<RouteConfig> = [
```

Comment (manuelojeda) [reply packages/@vue/cli-plugin-router/generator/template/src/router/index.js]: Adding this type annotation just bug the CLI and an error will popup

**Comment by manuelojeda** [comment]
I'm closing this PR to make a new one with the correct changes without adding the lock files.

**Pull request closed without merging**